### PR TITLE
Hyrax 357

### PR DESCRIPTION
**Pull request by vskorole**
**Summary:** Updated hyrax-metadata-updates with new OPeNDAP URL format.

Addresses [HYRAX-357:node.js development for Cumulus](https://bugs.earthdata.nasa.gov/browse/HYRAX-357?filter=-1)

## Changes

* Detailed list or prose of changes
* ...

## PR Checklist

- [x] Update CHANGELOG
- [x] Unit tests
- [x] Adhoc testing
- [ ] Integration tests

### CODE REVIEW

**Comment by Jkovarik** [comment]
Thanks for the contribution! I've let the team know it's here and someone will review it shortly.

File: tasks/hyrax-metadata-updates/index.js [hunk 1]
```diff
@@ -118,6 +118,55 @@ async function getEntryTitle(config, metadata, isUmmG) {
...
+  conceptId = get(result, '[0].id');
+  shortName = get(result, '[0].short_name');
+  versionId = get(result, '[0].version_id');
+  if (conceptId === undefined) {
+    throw new RecordDoesNotExist(`Unable to query parent collection using short name ${shortName} and version ${versionId}`);
+  }
+
+  return (config.addShortnameAndVersionIdToConceptId !== undefined
+         && config.addShortnameAndVersionIdToConceptId ===true) ?
+         conceptId + '/' + shortName + '.' + versionId : conceptId;
```
Comment (markdboyd) [inline tasks/hyrax-metadata-updates/index.js:167]: To ask a dumb question: using `conceptId + '/' + shortName + '.' + versionId` vs `conceptId` in the URL produces an equivalent response from CMR? If so, what is the purpose of adding this functionality to produce the former?

File: tasks/hyrax-metadata-updates/index.js [hunk 2]
```diff
@@ -133,8 +182,9 @@ async function generatePath(config, metadata, isUmmG) {
   if (providerId === undefined) {
     throw new InvalidArgument('Provider not supplied in configuration. Unable to construct path');
   }
-  const entryTitle = await getEntryTitle(config, metadata, isUmmG);
-  return `providers/${providerId}/collections/${entryTitle}/granules/${getGranuleUr(metadata, isUmmG)}`;
+  const entryCollection = await getCollectionEntry(config, metadata, isUmmG) ;
+  //return `providers/${providerId}/collections/${entryTitle}/granules/${getGranuleUr(metadata, isUmmG)}`;
```
Comment (markdboyd) [inline tasks/hyrax-metadata-updates/index.js:186]: Please remove this line

File: tasks/hyrax-metadata-updates/index.js [hunk 3]
```diff
@@ -133,8 +182,9 @@ async function generatePath(config, metadata, isUmmG) {
...
+  return `collections/${entryCollection}/granules/${getGranuleUr(metadata, isUmmG)}`;
```
Comment (markdboyd) [inline tasks/hyrax-metadata-updates/index.js:187]: Is it intentional that this path no longer begins with `providers/${providerId}/`?

File: tasks/hyrax-metadata-updates/index.js (hunk 1 above)
Comment (vskorole) [reply tasks/hyrax-metadata-updates/index.js:167]: https://wiki.earthdata.nasa.gov/display/DUTRAIN/Feature+analysis:+Restified+URL+for+OPENDAP+Data+Access

File: tasks/hyrax-metadata-updates/index.js (hunk 3 above)
Comment (vskorole) [reply tasks/hyrax-metadata-updates/index.js:187]: Yes.

File: tasks/hyrax-metadata-updates/index.js (hunk 2 above)
Comment (vskorole) [reply tasks/hyrax-metadata-updates/index.js]: Done

**Review by laurenfrederick — CHANGES_REQUESTED** [review]
@vskorole Thank you so much for this contribution and the test updates! Can you please add and entry to the `Changed` section in `CHANGELOG.md`?

File: tasks/hyrax-metadata-updates/schemas/config.json [hunk 4]
```diff
@@ -49,6 +49,11 @@
           "type": "string"
         }
       }
+    },
+    "addShortnameAndVersionIdToConceptId": {
+      "description": "Options for humanizing hyrax url",
```
Comment (laurenfrederick) [inline tasks/hyrax-metadata-updates/schemas/config.json:54]:
```suggestion
      "description": "Option to humanize the Hyrax URL. Humanizes when set to true.",
```

Comment (laurenfrederick) [reply tasks/hyrax-metadata-updates/schemas/config.json:54]: When do you not want to humanize?

**Comment by laurenfrederick** [comment]
Thanks @vskorole! Closing in favor of https://github.com/nasa/cumulus/pull/2136 so we can run integration tests and merge.

**Pull request closed without merging**